app/Models/Device.php

```diff
@@ -113,30 +114,14 @@ public static function findByHostname(string $hostname): ?Device
         return static::where('hostname', $hostname)->first();
     }
 
-    /**
-     * Returns IP/Hostname where polling will be targeted to
-     *
-     * @param  string|array  $device  hostname which will be triggered
-     *                                array  $device associative array with device data
-     * @return string IP/Hostname to which Device polling is targeted
-     */
-    public static function pollerTarget($device)
+    public function pollerTarget(): string
     {
-        if (! is_array($device)) {
-            $ret = static::where('hostname', $device)->first(['hostname', 'overwrite_ip']);
-            if (empty($ret)) {
-                return $device;
-            }
-            $overwrite_ip = $ret->overwrite_ip;
-            $hostname = $ret->hostname;
-        } elseif (array_key_exists('overwrite_ip', $device)) {
-            $overwrite_ip = $device['overwrite_ip'];
-            $hostname = $device['hostname'];
-        } else {
-            return $device['hostname'];
-        }
+        return $this->overwrite_ip ?: $this->hostname ?: '';
+    }
 
-        return $overwrite_ip ?: $hostname;
+    public function ipFamily(): AddressFamily
+    {
+        return str_ends_with($this->family ?? '', '6') ? AddressFamily::IPv6 : AddressFamily::IPv4;
```

Comment:
Should this be `$this->transport`:

https://community.librenms.org/t/fping-ipv6-only-devices-no-longer-working/28537/3?u=laf

Although testing this more generally and it looks like we might have always had a bug, when you don't have the fping6 binary and add an icmp only host, because the transport defaults to udp then it's always matches `AddressFamily::IPv4`